sceptre/hooks/cmd.py

```diff
@@ -19,10 +19,15 @@ def run(self):
         :raises: subprocess.CalledProcessError
         """
         envs = self.stack.connection_manager.create_session_environment_variables()
-        try:
-            subprocess.check_call(self.argument, shell=True, env=envs)
-        except TypeError:
-            raise InvalidHookArgumentTypeError(
-                'The argument "{0}" is the wrong type - cmd hooks require '
-                "arguments of type string.".format(self.argument)
-            )
+
+        if self.argument is None:
+            raise InvalidHookArgumentTypeError()
+
+        if isinstance(self.argument, str) or self.argument is None:
+            args = self.argument
+            executable = None
+        elif isinstance(self.argument, dict):
```

Comment:
passing in a dict is new so could you please add some docs for this?  an example would probably help as well.

also there should probably an else condition?  what if user passes in a list?  or you could probably keep the try/catch to handle all the other wacko things that users do.

Reply:
Indeed, I'll add some tests for invalid inputs so that I can handle those properly.